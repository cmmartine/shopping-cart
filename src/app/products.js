export const allProducts = [
  {
    prodName: 'Shoes',
    prodImg: 'Some image',
    prodDesc: 'Very good',
    prodCost: '$49.99',
    numInCart: 0,
    id: 1,
  },
  {
    prodName: 'Shirt',
    prodImg: 'Some image',
    prodDesc: 'Quality fabric',
    prodCost: '$19.99',
    numInCart: 0,
    id: 2,
  }
];