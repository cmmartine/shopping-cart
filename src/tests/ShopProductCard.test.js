import renderer from 'react-test-renderer';
import { screen } from '@testing-library/react';
import ShopProductCard from '../app/components/ShopProductCard';
import { renderWithProviders } from './test_utils/redux-store-test-util';
import { store } from '../app/store';
import { Provider } from 'react-redux';

describe('Shop Product Card', () => {
  const prop = {
    prodName: 'Shoes',
    prodImg: 'Some image',
    prodDesc: 'Very good',
    prodCost: '$49.99',
    numInCart: 0,
    d: 1,
  }

  it('renders correctly', () => {
    const view = renderer
      .create(
        <Provider store={store}>
          <ShopProductCard product={prop} />
        </Provider>
      )
      .toJSON();

      expect(view).toMatchSnapshot();
  });

  it('renders the card passed to it', () => {

    renderWithProviders(<ShopProductCard product={prop} />);

    expect(screen.getByRole('heading').textContent).toMatch('Shoes');
  });
});