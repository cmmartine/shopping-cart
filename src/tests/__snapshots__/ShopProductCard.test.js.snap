// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Product Card renders correctly 1`] = `
<div
  className="shop-card-inner"
>
  <img
    alt="Shoes"
    src="Some image"
  />
  <div
    className="prod-info-container"
  >
    <h3
      className="prod-name"
    >
      Shoes
    </h3>
    <p
      className="prod-desc"
    >
      Very good
    </p>
    <p
      className="prod-cost"
    >
      $49.99
    </p>
  </div>
  <div
    className="count-container"
  >
    <div>
      In cart: 
      0
    </div>
    <button
      aria-label="increment"
      className="shop-add-button"
      onClick={[Function]}
    >
      Add to Cart
    </button>
  </div>
</div>
`;
